File: problems/3005/solution.go
func maxFrequencyElements(nums []int) int {
    freq := make(map[int]int)
    best := 0
    for _, x := range nums {
        freq[x]++
        if freq[x] > best { best = freq[x] }
    }
    ans := 0
    for _, count := range freq {
        if count == best { ans += count }
    }
    return ans
}

File: problems/3005/solution.c
int maxFrequencyElements(int* nums, int numsSize) {
    int freq[101] = {0};
    int best = 0;
    for (int i = 0; i < numsSize; ++i) {
        int count = ++freq[nums[i]];
        if (count > best) best = count;
    }
    int ans = 0;
    for (int x = 1; x <= 100; ++x)
        if (freq[x] == best) ans += freq[x];
    return ans;
}
